lib/Onnxifi/base.h

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_ONNXIFI_BASE_H
+#define GLOW_ONNXIFI_BASE_H
+
+#include "glow/ExecutionEngine/ExecutionEngine.h"
+
+#include "onnx/onnxifi.h"
+
+#include <mutex>
+
+namespace glow {
+namespace onnxifi {
+
+/// BackendId associated with the Glow backend.
+class BackendId {
+public:
+  BackendId(int id) : id_(id) {}
+
+private:
+  int id_;
+};
+
+typedef BackendId *BackendIdPtr;
+
+class Backend {
+public:
+  Backend(BackendIdPtr backendId) : backendIdPtr_(backendId) {}
+
+private:
+  BackendIdPtr backendIdPtr_;
+};
+
+typedef Backend *BackendPtr;
+
+class Event {
+public:
+  Event() : fired{false} {}
+  /// Signal
+  bool signal();
+  /// Wait
+  void wait();
+
+private:
+  bool fired;
+  std::mutex mutex_;
+  std::condition_variable cond_;
+};
+
+typedef Event *EventPtr;
+
+class Graph {
+public:
+  Graph(BackendPtr backendPtr) : backendPtr_(backendPtr) {}
+
+  /// InitGraph
+  onnxStatus initGraph(const void *onnxModel, size_t onnxModelSize,
+                       uint32_t weightCount,
+                       const onnxTensorDescriptor *weightDescriptors);
+  /// Set IO
```

Comment:
Period.

Reply:
yeah, those comments are not finalized. I'll fix all those for better documentation.